learn/configuration/instance_options.md

```diff
@@ -192,17 +192,17 @@ Sets the HTTP address and port Meilisearch will use.
 **Environment variable**: `MEILI_MASTER_KEY`
 **CLI option**: `--master-key`
 **Default value**: `None`
-**Expected value**: an alphanumeric string
+**Expected value**: UTF-8 string of at least 16 bytes
 
 Sets the instance's master key, automatically protecting all routes except [`GET /health`](/reference/api/health.md). This means you will need an API key to access endpoints such as `POST /search` and `GET /documents`. [You can read more about security keys in Meilisearch in our dedicated guide.](/learn/security/master_api_keys.md)
 
-::: note
-You must supply an alphanumeric string when using this option.
-:::
+Providing a master key is mandatory when `--env` is set to `production`. If none is given, or it is less than 16 bytes, Meilisearch will throw an error and refuse to launch. In both cases, Meilisearch will suggest a secure autogenerated master key you can use by running the following command and restarting your instance:
 
-Providing a master key is mandatory when `--env` is set to `production`. If none is given, Meilisearch will throw an error and refuse to launch.
+```
+--master-key="SUGGESTED_MASTER_KEY"
+```
 
-If no master key is provided in a `development` environment, all routes will be unprotected and publicly accessible.
+If no master key is provided in a `development` environment, all routes will be unprotected and publicly accessible. Meilisearch will suggest an autogenerated secure master key if you need protection in a `development` environment.
```

Comment:
Since autogenerated key is supplied no matter the environment, I'd suggest we put this info in a separate paragraph instead of repeating content.

```suggestion
If no master key is provided in a `development` environment, all routes will be unprotected and publicly accessible.

If you do not supply a master key in `production` or `development` environments, Meilisearch will suggest a secure autogenerated master key you can use when restarting your instance.
```